core/samples/net/udp/main.cpp

```diff
@@ -0,0 +1,79 @@
+#include <cstring>
+#include <iostream>
+#include <thread>
+
+#include <cubos/core/log.hpp>
+/// [Import header]
+#include <cubos/core/net/udp_socket.hpp>
+/// [Import header]
+#include <cubos/core/reflection/external/cstring.hpp>
+#include <cubos/core/reflection/external/primitives.hpp>
+
+using cubos::core::net::Address;
+using cubos::core::net::UdpSocket;
+
+/// [Define constants]
+#define SERVER_ADDRESS Address::LocalHost
+#define SERVER_PORT 8080
+
+#define CLIENT_ADDRESS Address::LocalHost
+#define CLIENT_PORT 8081
+/// [Define constants]
+
+/// [Server function]
+void runServer()
+{
+    UdpSocket server;
+    CUBOS_ASSERT(server.bind(SERVER_PORT, SERVER_ADDRESS), "Failed to bind server socket");
+
+    CUBOS_INFO("Server is running and waiting for messages at port {}", SERVER_PORT);
+
+    char buffer[1024];
+    size_t received;
+    Address senderAddress;
+    while (true)
+    {
+        if (server.receive(buffer, sizeof(buffer), received, senderAddress))
+        {
+            buffer[received] = '\0'; // C strings :)
+            CUBOS_INFO("Received message: {} with size {}", buffer, received);
+            break;
+        }
+
+        CUBOS_ERROR("Receive error");
+    }
+}
+/// [Server function]
+
+/// [Client function]
+void runClient()
+{
+    UdpSocket client;
+    CUBOS_ASSERT(client.bind(CLIENT_PORT, CLIENT_ADDRESS), "Failed to bind client socket");
```

Comment:
This line should not be necessary. You don't need to bind a UDP socket to send a packet, you only need to bind it to receive one.